corehq/apps/reports/standard/deployments.py

```diff
@@ -522,14 +522,18 @@ def _get_formatted_assigned_location_names(self, user, user_loc_dict):
 
         formatted_str = ', '.join(formatted_loc_names[:4])
         all_str = ', '.join(formatted_loc_names)
-        eplise_str = _('...See more') if len(formatted_loc_names) > 4 else ''
+        view_all_str = _('...See more') if len(formatted_loc_names) > 4 else ''
+        view_less_str = _('...See less') if len(formatted_loc_names) > 4 else ''
         out_str = ('''
             <div>
                 <span class="locations-list">{}</span>
-                <a href="#" class="toggle-all-locations">{}</a>
                 <span class="all-locations-list" style="display:none">{}</span>
+                <a href="#" class="toggle-all-locations">
+                   <span>{}</span>
+                   <span style="display:none">{}</span>
+                </a>
             </div>
-        ''').format(formatted_str, eplise_str, all_str)
+        ''').format(formatted_str, all_str, view_all_str, view_less_str)
```

Comment:
I think this code will be easier to follow if we add additional html elements later. And we are just adding empty nodes which can be avoided.
something like:
```
html_nodes = ['<span class="locations-list">{formatted_str}</span>']
if len(formatted_loc_names) > 4:
    html_nodes.append[
        '<span class="all-locations-list" style="display:none">{all_str}</span>',
        '...'
    ]
return format_html(f'<div>{''.join(html_nodes)}</div>')
```

Reply:
I really like this approach, a bit cleaner. I've done some refactoring in ([ee88a55](https://github.com/dimagi/commcare-hq/pull/34744/commits/ee88a5567458320e630905c6d2f6a3d1048cbfc0)).